src/main/java/com/gtnewhorizon/structurelib/structure/IStructureDefinition.java

```diff
@@ -324,9 +324,6 @@ static <T> boolean iterate(
             int basePositionC,
             boolean hintsOnly,
             Boolean checkBlocksIfNotNullForceCheckAllIfTrue) {
-        if (world.isRemote ^ hintsOnly) {
```

Comment:
Make it return false if `!world.isRemote && hintsOnly`. Makes no sense to generate hints on server side.